### PR TITLE
Fix BatchEncoding not prepending batch axis when tensor_type is None

### LINKED ISSUE #28882: `BatchEncoding` won't prepend batch axis to python list, even `prepend_batch_axis` is `True`
### System Info

- `transformers` version: 4.35.2
- Platform: Linux-6.5.0-14-generic-x86_64-with-glibc2.35
- Python version: 3.11.6
- Huggingface_hub version: 0.20.2
- Safetensors version: 0.4.1
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.1.2+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
>>> BatchEncoding({'input_ids': [1]}, prepend_batch_axis=True)
{'input_ids': [1]}
```

### Expected behavior

Output should be:
```
{'input_ids': [[1]]}
```

### PR DESCRIPTION
**Pull request by mig-mfreitas**
# Fix BatchEncoding not prepending batch axis to python list when tensor_type is None

Fixes #28882

## Who can review?
@ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Maybe defaulting `prepend_batch_axis` to False should help. We can gradually change what's returned but I think the fast also returns no batch no? 
image

For a simple T5 we already have a fairly strange API but people are used to it 😓
